warcraft/contracts/assets/stateful.py

```diff
@@ -0,0 +1,62 @@
+from pyteal import *
+
+def approval_program():
+    app_manager = Bytes("app_manager")
+    total_registered = Bytes("total_registered")
+    escrow = Bytes("escrow")
+    # verify rekey_to and close_rem_to are set as zero_address
+    basic_checks = And(
+        # Always verify that the RekeyTo property of any transaction is set to the ZeroAddress
+        # unless the contract is specifically involved in a rekeying operation.
+        Txn.rekey_to() == Global.zero_address(),
+        Txn.close_remainder_to() == Global.zero_address(),
+        Txn.asset_close_to() == Global.zero_address()
```

Comment:
I think this should be `Gtxn[0]`.

Reply:
why ? this is for transaction sent to app and we use Txn for that

Reply:
this should be moved to escrow contract